sparrow-py/pysrc/sparrow_py/_timestream.py

```diff
@@ -952,6 +952,36 @@ def cast(self, data_type: pa.DataType) -> Timestream:
         """
         return Timestream(self._ffi_expr.cast(data_type))
 
+    def seconds_since(self, time: Union[Timestream, datetime]) -> Timestream:
+        """
+        Create a Timestream containing the number of seconds since `time`.
+
+        Parameters
+        ----------
+        time : Union[Timestream, datetime]
+            The time to compute the seconds since.
+            This must be a datetime or a Timestream of timestamp_ns.
+
+        Returns
+        -------
+        Timestream
+            Timestream containing the number of seconds since `time`.
+        """
+        if isinstance(time, datetime):
+            session = self._ffi_expr.session()
+            nanos = Timestream._literal(time.timestamp() * 1e9, session=session)
+            nanos = Timestream.cast(nanos, pa.timestamp('ns', None))
+            self_nanos = Timestream._call("time_of", self)
```

Comment:
Based on the conversation, it sounds like this shouldn't call `time_of` -- it should expect `self` to be a timestream of timestamps.

Reply:
Perhaps in the demo, though I was intending for this function to be called on any streaming event. 

Essentially just syntactic sugar that skips you having to get the time column out and call lag yourself.

Reply:
Ope, I'm confusing myself with the other PR.